components/Spedizione.js

```diff
@@ -58,67 +77,104 @@ function Spedizione({ name, state, handle,boolean, allInfo, setDataToStore}) {
           <hr />
           <div className="master-children-partenza">
             <div className="children-partenza">
-              <label className="control-label">Nome</label>
+              <form  onSubmit={(e)=>prevent(e)} >
+              <label htmlFor="nameS" className="control-label">Nome</label>
               <input
                 placeholder=""
+                maxLength="20"
                 className="form-control"
                 defaultValue=""
-                onChange={(e) => handle(e, "nameDispatcher")}
+                type="text"
+                required
+                id="nameS"
+                onChange={(e) => {handle(e, "nameDispatcher");valdationFunction("nameS","butNaS")}}
```

Comment:
in genere le validation functions dovrebbero essere chiamate prima di fare qualsiasi altra azione sull'input.

Questo tipo di espressione nella callback e' poco chiaro, inserisci la validationFunction all'interno dell'handle. Idealmente una callback dovrebbe essere quanto piu semplice possibile

`onChange={(e)=>handleChange(e, 'name')}`